### PR TITLE
sys/event: add event_wait_timeout_ztimer()

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
If one wants to use the timeout functionality with the `event` module, it is so far tied to `xtimer`. This PR also adds support to use `ztimer` for the same purpose, by introducing a new function called `event_wait_timeout_ztimer()`. This function allows to specify the used ztimer clock, and hence it can be used for any fitting time scale and no 64-bit version of that function is needed.

I'd like to have this function merged to proceed on my path to remove any `xtimer` and `periph_timer` dependencies from NimBLE...

### Testing procedure
Run the `tests/event_wait_timeout` test on any platform of your choosing. It should still succeed.

### Issues/PRs references
none

### CODE REVIEW

File: sys/event/event.c [hunk 1]
```diff
@@ -106,30 +106,57 @@ static event_t *_wait_timeout(event_queue_t *queue, xtimer_t *timer)
...
+#endif
+
+#if IS_USED(MODULE_XTIMER)
+static event_t *_wait_timeout_xtimer(event_queue_t *queue, xtimer_t *timer)
+{
+    event_t *result = _wait_timeout(queue);
     if (result) {
         xtimer_remove(timer);
     }
-
```
Comment (fjmolinas) [inline sys/event/event.c:116]: Unrelated, is this related to uncrustifying the file? If so can you split into its own commit?

File: tests/event_wait_timeout/Makefile [hunk 2]
```diff
@@ -3,6 +3,7 @@ include ../Makefile.tests_common
 FORCE_ASSERTS = 1
 USEMODULE += event
 USEMODULE += xtimer
+USEMODULE += ztimer ztimer_usec
```
Comment (fjmolinas) [inline tests/event_wait_timeout/Makefile:6]: This change means that `xtimer_on_ztimer` is pulled in so the whole test now runs on `ztimer`, not sure how to proceed here, in a lot of cases `ztimer` is not tested unless explicitly pulled in, but in this case, maybe the best is having an individual test?

**Review by fjmolinas — COMMENTED** [review]
Looks good, only a comment on how the test is integrated, see below.

**Comment by kfessel** [comment]
`event_wait_timeout` may be `provided using event_wait_timeout_ztimer()` and `ZTIMER_USEC`

**Comment by fjmolinas** [comment]
ping @haukepetersen !

**Comment by fjmolinas** [comment]
ping @haukepetersen!

**Comment by haukepetersen** [comment]
Sorry for the delay, had my head stuck somewhere else :-)

Did address the open comments:
- redone that newline change
- added a separate test application for testing `event_timeout_ztimer`. Now the original test application is untouched and still solely depending on xtimer, while the new application tests the ztimer aspects.

**Comment by fjmolinas** [comment]
Ran the test on many BOARD's all passing: https://ci.inria.fr/ci-riot-tribe/job/build-pipeline.jk/259/artifact/

**Review by fjmolinas — APPROVED** [review]
ACK.
